### PR TITLE
Sed

### PR DESCRIPTION
**Pull request by joe-getcouragenow**
added /internal/download that is cross platform
- See .gitignore to see where it puts things on the OS.

Added makefile.
- go:generate was not working on Mac. makefile worked first time. Hope this is ok. On Windows and Linux make files work these days. So we can use the makefile later with github actions.
-  The LIB_FSPATH is just  there for helping with the github actions later !

When i do "make build", everything works.

When  i do "make server-run", i get some errors. I think fixing those errors is the next stage to work on.

### CODE REVIEW

**Review by ctessum — CHANGES_REQUESTED** [review]
Thanks! I think the main issue before we merge this is that `index.html` has been changed, where before it included everything needed to display the GUI, but now it looks like the default WASM demo page.

**Comment by joe-getcouragenow** [comment]
> Thanks! I think the main issue before we merge this is that `index.html` has been changed, where before it included everything needed to display the GUI, but now it looks like the default WASM demo page.

Ok. Index.html fixed and:
- I commented out line 74 in the make file to stop stomping on the index.html
```  cp $(GOROOT)/misc/wasm/wasm_exec.html $(LIB_FSPATH)/gui/html/index.html ``` 
- I missed the fact that your also commented it out:
``` // //go:generate bash -c "cp $DOLLAR(go env GOROOT)/misc/wasm/wasm_exec.html ./gui/html/index.html" ````


---

Also here is a simple example of the concept of getting the Github Actions to use the makefile: https://www.freecodecamp.org/news/a-lightweight-tool-agnostic-ci-cd-flow-with-github-actions/
- I have used this before because its keeps things agnostic but gets your LIN, MAC and WIN builds and testing, so you know it works for everyone. 
- I think that doing this next makes sense, so we can validate that this approach works for all Os's.

File: internal/download/download-protoc.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		fmt.Println("You are running on Windows")
+		protocZipFilename = ProtocZipFilenameWIN
+	}
+
+	if runtime.GOOS == "darwin" {
+		fmt.Println("You are running on Mac")
+		protocZipFilename = ProtocZipFilenameMAC
+	}
+
+	if runtime.GOOS == "lunix" {
```
Comment (ctessum) [inline internal/download/download-protoc.go]: lunix -> linux

**Review by ctessum — CHANGES_REQUESTED** [review]

File: internal/download/download-protoc.go (hunk 1 above)
Comment (joe-getcouragenow) [reply internal/download/download-protoc.go]: thanks - will fix now.

**Comment by joe-getcouragenow** [comment]
I just saw that Travis CI failed.
It is probably setup to use the go generate approach.
Also i see it pulls in your other project using github.com/spatialmodel/inmap/cmd/inmap

My feeling is that we should change over to using github actions and then cal the make file from it now.
So if you try the make file from your linux computer now, and if that works then the next thing to do is the github actions.

**Pull request merged**